src/serializationStrategies/XpathRangeSelector/xpath.spec.js

```diff
@@ -297,14 +309,28 @@ describe('getXPathForElement', () => {
 
   it('doesn\'t count highlights between previous text', () => {
     document.body.innerHTML = `
-      <div id="reference">qwer<span ${DATA_ATTR}>asdf</span><div id="target"></div><span ${DATA_ATTR}>as</span>asdfasdf</div>
+      <div id="reference">` +
+        `qwer` +
+        `<span ${DATA_ATTR}>` +
+          screenReaderNode +
+          `asdf` +
+          screenReaderNode +
+        `</span>` +
+        `<div id="target"></div>` +
+        `<span ${DATA_ATTR}>` +
+          screenReaderNode +
+          `as` +
+          screenReaderNode +
+        `</span>` +
+        `asdfasdf` +
+      `</div>
     `;
 
     const reference = document.getElementById('reference');
     const [result, offset] = xpath.getXPathForElement(reference, 4, reference);
 
     expect(result).toEqual("./text()[2]");
-    expect(offset).toEqual(2);
+    expect(offset).toEqual(1);
```

Comment:
I don't know if this change is correct

Reply:
xpaths use indexes starting at 1 but looking at this text i assume the input to `getXPathForElement` has more to do with the node offset in javascript which is 0 indexed. so our starting point for `xpath.getXPathForElement(reference, 4, reference);` is the 4th index (starting at 0) of `div#reference`, which is the last text node, so the "cursor" is between the `as` and the `asdfasdf`.

after ignoring highlights the 4 text nodes become two text nodes separated by `div#target`, the result is `./text()[2]` (remember offsets starting at 1 now) so thats the second text node, which is right, but the cursor should be after the `as`, so i think the `2` in the original test is correct

Reply:
I've updated how adjusting offset works and now it passes with `2`:
https://github.com/openstax/highlighter/pull/42/files#diff-fabc66fa1eab8f3774c4988fb0f16f24b134fef2bcc5a4d9f43cebea513d7642R122